src/components/ProblemsTable.tsx

```diff
@@ -68,58 +79,124 @@ export const ProblemsTable = ({
     [userFileTypeIcons]
   );
 
-  const expandReports = (reports: Report[]): { category: string; subRows: Report[] }[] => {
-    const expandableReports: Record<string, Report[]> = {};
+  const getFileNameFromId = React.useCallback(
+    (id?: string) => {
+      const filesInfo = sourceFilesInfo || context?.reportData.sourceFilesInfo;
+      return filesInfo?.fileId === id
+        ? filesInfo?.fileName
+        : filesInfo?.Files?.find((file) => file.fileId === id)?.fileName;
+    },
+    [sourceFilesInfo, context?.reportData.sourceFilesInfo]
+  );
+
+  const fileData = React.useMemo(() => {
+    const filesInfo = sourceFilesInfo || context?.reportData.sourceFilesInfo;
+    return [
+      { ...filesInfo, mainFile: true },
+      ...(filesInfo && filesInfo.Files ? filesInfo.Files.filter((file: SourceFile) => file.state !== 'Hidden') : []),
```

Comment:
we could have defined states as a type too, now we check state with plain text

Reply:
this part of the method was copied from the old FilesTable.tsx module. The 'SourceFile' types' 'state' property is set to the primitive type 'string' [in Report.tsx]; 
This could be replaced by a set of string literal types, but we would need to be sure what those could be defined as, and that change would be better in its own PR